data-reconciliation-app/src/endpoints/reporting.ts

```diff
@@ -0,0 +1,31 @@
+import * as ccfapp from "@microsoft/ccf-app";
+import { ServiceResult } from "../utils/service-result";
+import { reportingService, authenticationService } from "../utils/dependencies";
+import { ApiResult } from "../utils/api-result";
+
+export function getDataSummary(
+  request: ccfapp.Request<any>
+): ccfapp.Response<any> {
+  try {
+    const getCallerId = authenticationService.getCallerId(request);
+    if (getCallerId.failure) {
+      return ApiResult.Response(getCallerId);
+    }
+
+    const callerId = getCallerId.content;
+    const isMember = authenticationService.isMember(callerId);
```

Comment:
Should we use isUser?

Reply:
I've modified it to accept calls from both members and users; later, I will change the test cases to test both scenarios.